### PR TITLE
Remove old feature switches

### PR DESCRIPTION
**Pull request by ianwjhalliday**
Addresses #6.  Though there are a few more switches remain that I believe would be good to remove.

Removes the following switches:
 - `LetConst`
 - `Map`
 - `Set`
 - `WeakMap`
 - `ES6WeakSet`
 - `DefineGetterSetter`
 - `__proto__`
 - `ES6Iterators`
 - `ES6Lambda`
 - `ES6NumericLiteral`
 - `ES6StringTemplate`
 - `ES6Symbol`
 - `TDZ`

With the removal of `-LetConst` a couple of predicates become always true and have been removed. They are:
 - `BindDeferredPidRefs()`
 - `IsBlockScopeEnabled()`

Parser binding had specific logic to handle catch scopes in non-block scope mode to support legacy ES5 behavior.  This has also been removed.

Finally `Parser::CheckStrictModeFncDeclNotSourceElement()` no longer did anything so it has also been removed.

### CODE REVIEW

**Comment by ianwjhalliday** [comment]
@pleath appreciate you taking a look, the last few commits in particular.

FYI @tcare @boingoing @aneeshdk @akroshg @suwc @goyakin @abchatra

File: lib/Runtime/Library/JavascriptLibrary.cpp [hunk 1]
```diff
@@ -3536,17 +3391,8 @@ namespace Js
...
-        {
-            propertyCount += 1;
-        }
-        if (scriptContext->GetConfig()->IsES6ObjectExtensionsEnabled())
-        {
-            propertyCount += 2;
-        }
 
-        typeHandler->Convert(objectConstructor, mode, propertyCount);
+        typeHandler->Convert(objectConstructor, mode, 20);
```
Comment (pleath) [inline lib/Runtime/Library/JavascriptLibrary.cpp]: Can we seize the opportunity to replace "20" with a meaningfully-named constant?

File: lib/Runtime/Library/JavascriptLibrary.cpp [hunk 2]
```diff
@@ -3603,9 +3443,7 @@ namespace Js
         JavascriptLibrary* library = objectPrototype->GetLibrary();
         ScriptContext* scriptContext = objectPrototype->GetScriptContext();
 
-        bool hasAccessors = scriptContext->GetConfig()->Is__proto__Enabled();
-
-        typeHandler->Convert(objectPrototype, mode, 11, hasAccessors);
+        typeHandler->Convert(objectPrototype, mode, 11, true);
```
Comment (pleath) [inline lib/Runtime/Library/JavascriptLibrary.cpp]: Can "11" get a name also?

File: lib/Runtime/Library/JavascriptLibrary.cpp (hunk 1 above)
Comment (ianwjhalliday) [inline lib/Runtime/Library/JavascriptLibrary.cpp]: I considered that but all these Initialize functions use unnamed constants, and when you look at the parameter name its obvious what the value is for.  I decided it wasn't worth bothering with.

**Comment by abchatra** [comment]
It would be nice to keep -ES6Promise,  ES6Unscopables, ES6Number and ES6Object switches. These API's are frequently overridden by various polyfill's. An easy way to disable these comes in handy for website reductions.

**Comment by ianwjhalliday** [comment]
@abchatra good catch.  I'll update the PR to leave out those removals.

File: lib/Runtime/Library/ObjectPrototypeObject.cpp [hunk 3]
```diff
@@ -8,7 +8,7 @@ namespace Js
 {
     ObjectPrototypeObject::ObjectPrototypeObject(DynamicType* type) : DynamicObject(type)
     {
-        __proto__Enabled = GetScriptContext()->GetConfig()->Is__proto__Enabled();
+        __proto__Enabled = true; // TODO[ianhall]: Does this still apply if __proto__ is now always enabled? Check with JC
```
Comment (pleath) [inline lib/Runtime/Library/ObjectPrototypeObject.cpp]: I wonder if we can now eliminate the flag altogether.

**Comment by abchatra** [comment]
Any update here?

**Comment by abchatra** [comment]
Any update here?

**Comment by ianwjhalliday** [comment]
@dotnet-bot test Windows x64_debug please

**Pull request merged**